### DISCUSSION TITLE
Segurança em APIs ASP.NET com JWT e Bearer Authentication - Referencias entre os projetos

### DISCUSSION THREAD

**Discussion by KIQxl** [balta-io/forum#634] [Tira Dúvidas] [answered]
No curso Segurança em APIs durante a aula de configuração da API, é adicionada a referencia do projeto de infra no projeto da API, essa prática é recomendada ou existe alguma forma melhor para trabalhar com os projetos?
de qualquer forma é necessário adicionar o contexto na classe program.cs do projeto de API, porém, a aplicação não fica mais exposta e suscetível a ataques quando minha camada de API tem referência a camada de infra que faz o acesso aos dados?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @KIQxl, como vai?

No curso Segurança em APIs durante a aula de configuração da API, é adicionada a referencia do projeto de infra no projeto da API, essa prática é recomendada ou existe alguma forma melhor para trabalhar com os projetos?
- Sim, esta é uma das práticas utilizadas

de qualquer forma é necessário adicionar o contexto na classe program.cs do projeto de API, porém, a aplicação não fica mais exposta e suscetível a ataques quando minha camada de API tem referência a camada de infra que faz o acesso aos dados?
- A API roda no servidor, ninguém tem acesso (Ou não deveria) ao código... tem um exemplo para me dar?

**Comment by KIQxl** [comment]
Bom dia, @andrebaltieri !
Na verdade, essa dúvida era referente a arquitetura mesmo, qual a melhor forma de relacionar os projetos e organizar a aplicação. Por isso a pergunta anterior. Era mais para consolidar a melhor forma de relacionar os projetos. 

Obrigado!
